scripts/filter_type.js

```diff
@@ -0,0 +1,82 @@
+//used for the filter type to be determined once x_axis and y_axis are selected
+
+$(document).ready(function() {
+    var plots = [];
+
+    $("#y-axis-search, #x-axis-search").change(function() {
+        $("#filter-type").empty().append('<option disabled selected value> -Filter- </option>')
+        $.getJSON(json_url, function(plots) {
+
+            var y_axis = $("#y-axis-search");
+            var x_axis = $("#x-axis-search");
+            var dist_options = [];
+
+            //collects all the distinct values of each sort_option feature in the json
+            for (var y = 0; y < sort_options.length; y++) {
+                var lookup = {};
+                var result = [];
+
+                for (var item, i = 0; item = plots[i++];) {
+                    var name = item[sort_options[y]];
+
+                    if (!(name in lookup)) {
+                        lookup[name] = 1;
+                        result.push(name);
+                    }
+                }
+                result.sort();
+                dist_options.push(result);
+            }
+            //dist_options includes distinct values of sort options as seperate arrays of strings
+
+            $("#filter-type").empty().append('<option disabled selected value> -Filter- </option>');
```

Comment:
This is being called up above, why is it still being called here? Seems like you should be able to consolidate/simplify this.